File: src/main/java/xyz/terrific/client/ClientMain.java
package xyz.terrific.client;

import java.net.Socket;

public enum ClientMain {
    INSTANCE;

    public void init(String ip, int port) throws Exception {
        Socket socket = new Socket(ip, port);
        Client client = new Client(socket);
        client.start();

    }
    public void init(String name, String ip, int port) throws Exception {
        Socket socket = new Socket(ip, port);
        Client client = new Client(socket, name);
        client.start();

    }

}

File: src/main/java/xyz/terrific/client/ClientMain.kt
package xyz.terrific.client

import java.net.Socket

enum class ClientMain {
    INSTANCE;

    @Throws(Exception::class)
    fun init(ip: String?, port: Int) {
        val socket = Socket(ip, port)
        val client = Client(socket)
        client.start()
    }

    @Throws(Exception::class)
    fun init(name: String?, ip: String?, port: Int) {
        val socket = Socket(ip, port)
        val client = Client(socket, name)
        client.start()
    }
}
